sig-architecture/production-readiness.md

```diff
@@ -15,63 +15,121 @@ aspects of the process.
+  Describe them providing:
+  - API type
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+
+* **Will enabling / using this feature result in any new calls to cloud
+  provider?**
+
+* **Will enabling / using this feature result in increasing size or count
+  of the existing API objects?**  
+  Describe them providing:
+  - API type(s):
+  - Estimated increase in size: (e.g. new annotation of size 32B)
+  - Estimated amount of new objects: (e.g. new Object X for every existing Pod)
+
+* **Will enabling / using this feature result in increasing time taken by any
+  operations covered by [existing SLIs/SLOs][]?**  
+  Think about adding additional work or introducing new steps in between
+  (e.g. need to do X to start a container), etc. Please describe the details.
+
+* **Will enabling / using this feature result in non-negligible increase of
+  resource usage (CPU, RAM, disk, IO, ...) in any components?**  
+  Things to keep in mind include: additional in-memory state, additional
+  non-trivial computations, excessive access to disks (including increased log
+  volume), significant amount of data send and/or received over network, etc.
+  This through this both in small and large cases, again with respect to the
+  [supported limits][].
+
+#### Rollout, Upgrade and Rollback Planning
+
+#### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**  
+  Think about both cluster-level services (e.g. metrics-server) as well
+  as node-level agents (e.g. specific version of CRI).
+
+* **How does this feature respond to complete failures of the services on which
+  it depends?**  
+  Think about both running and newly created user workloads as well as
+  cluster-level services (e.g. DNS).
+
+* **How does this feature respond to degraded performance or high error rates
+  from services on which it depends?**
+
+#### Monitoring requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+* **How can an operator determine if the feature is functioning properly?**  
+  Focus on metrics that cluster operators may gather from different
+  components and treat other signals as last resort.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+  - [ ] Metrics
+	  - Metric name:
+    - [Optional] Aggregation method:
+    - Components exposing the metric:
+  - [ ] Other (treat as last resort)
+    - Details:
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+#### Thoubleshooting
+
+* **What are the known failure modes?**
+
+* **How can those be detected via metrics or logs?**
+
+* **What are the mitigations for each of those failure modes?**
+
+* **What are the most useful log messages and what logging levels to they require?**  
+  TODO(wojtek-t): Is that really a question that makes sense at the KEP level?
```

Comment:
No, this is more of a playbook question. We could instead request a playbook at alpha -> beta or beta -> GA, and move troubleshooting there, perhaps along with monitoring details. Or, if we want to keep everything in the KEP then we can make the troubleshooting section only required upon beta -> GA.

Reply:
Makes sense - added a comment about that.